integration-tests/reactive-pg-client/src/main/java/io/quarkus/it/reactive/pg/client/FruitResource.java

```diff
@@ -34,7 +34,7 @@ void setupDb() {
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     public CompletionStage<JsonArray> listFruits() {
-        return client.query("SELECT * FROM fruits").thenApply(pgRowSet -> {
+        return client.query("SELECT * FROM fruits").thenApplyAsync(pgRowSet -> {
```

Comment:
We don't recommend using async without executor as it may behave very differently in containers. But as it's an IT, should not matter too much.

Reply:
Errr, didn't we add `Async` to work around the fact that we're on the IO thread? I thought this PR fixed that issue by getting around it. @stuartwdouglas are you sure about this line?

Reply:
That came from the revert

Reply:
Well, ok, but that was a workaround that the code reverted fixed. Your new dual-vertx also fixes the issue so that workaround is also not needed anymore.